lib/IMAP/LazyHordeImapClient.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\IMAP;
+
+use Horde_Imap_Client_Socket;
+use OCA\Mail\Account;
+
+class LazyHordeImapClient {
```

Comment:
Isn't the horde client always a lazy client, because it only logs in on demand?

Reply:
True, I guess the logic can be simplified to just create a single client for the chain and pass it around. The `getClient()` method wasn't sampled once so the factory overhead is negligible.